### DISCUSSION TITLE
Add default state for orders

### DISCUSSION THREAD

**Discussion by JustasBelevicius** [PrestaShop/PrestaShop#40220] [Ideas]
**Is your feature request related to a problem?**
Many stores have different order status flow (for example, a store might want to check if products are actually in a certain location before approving the request).

**Describe the solution you'd like**
The simplest solution is to add an input field which allows to select a default order state when the order is placed. A more involved solution could be a graph representation of order flow allowing to create multiple nodes (order states) and describing transitions between those states.

**Describe alternatives you've considered**
For now I've overwritten payment modules validate.php to use a hardcoded int value.

**Comment by JustasBelevicius** [comment]
This would require payment module developers to use PS_DEFAULT_ORDER_STATE variable in their code.

**Comment by khouloudbelguith** [comment]
Hi @JustasBelevicius,

Thanks for your report.
Ping @PrestaShop/prestashop-product-team what do you think?
Thanks!
